core/src/main/java/org/apache/accumulo/core/lock/ServiceLockData.java

```diff
@@ -164,30 +165,22 @@ public ServiceLockData(UUID uuid, String address, ThriftService service) {
 
   public String getAddressString(ThriftService service) {
     ServiceDescriptor sd = services.get(service);
-    if (sd == null) {
-      return null;
-    }
-    return sd.getAddress();
+    return sd == null ? null : sd.getAddress();
   }
 
   public HostAndPort getAddress(ThriftService service) {
-    return AddressUtil.parseAddress(getAddressString(service), false);
+    String s = getAddressString(service);
+    return s == null ? null : AddressUtil.parseAddress(s, false);
```

Comment:
This is the main change that fixes things in this PR. The other changes in this class were just taking the opportunity to make the other methods that do the same thing a bit more concise and consistent with one another.

Reply:
Leveraging the existing optional and replacing if stmts with Optional.map stmts was a really nice change.

Reply:
It wasn't strictly necessary to do that to fix the issue, but it made the code so much easier to understand while I was trying to track down and verify there were no negative side-effects by any of the existing callers when I changed the behavior of this getAddress method.